host_core/native/hostcore_wasmcloud_native/src/client.rs

```diff
@@ -43,30 +43,67 @@ impl TokenManager for PickYourAuth {
     }
 }
 
-/// Returns a bindle client configured to cache to disk
-pub async fn get_client() -> Result<CachedClient, Box<dyn std::error::Error + Sync + Send>> {
-    let auth = if let Ok(pw) = var(BINDLE_PASSWORD_ENV) {
-        if let Ok(username) = var(BINDLE_USER_NAME_ENV) {
-            PickYourAuth::Http(HttpBasic::new(&username, &pw))
-        } else {
-            return Err(
-                "Bindle password was set, but no username was given. Unable to configure client"
-                    .into(),
-            );
+fn get_bindle_auth(creds_override: Option<HashMap<String, String>>) -> PickYourAuth {
+    if let Some(co) = creds_override {
+        match (co.get("username"), co.get("password"), co.get("token")) {
+            (Some(u), Some(p), _) => PickYourAuth::Http(HttpBasic::new(u, p)),
+            (_, _, Some(t)) => PickYourAuth::LongLived(LongLivedToken::new(t)),
+            _ => PickYourAuth::None(NoToken),
         }
-    } else if let Ok(token) = var(BINDLE_TOKEN_ENV) {
-        PickYourAuth::LongLived(LongLivedToken::new(&token))
     } else {
-        PickYourAuth::None(NoToken)
-    };
+        match (
+            var(BINDLE_PASSWORD_ENV),
+            var(BINDLE_USER_NAME_ENV),
+            var(BINDLE_TOKEN_ENV),
+        ) {
+            (Ok(pw), Ok(username), _) => PickYourAuth::Http(HttpBasic::new(&username, &pw)),
+            (_, _, Ok(token)) => PickYourAuth::LongLived(LongLivedToken::new(&token)),
```

Comment:
After looking at https://github.com/deislabs/bindle/blob/main/docs/README.md#configuring-authentication, does Bindle even support just token authentication? I see github oauth2 and http basic auth. cc @thomastaylor312

Reply:
the `LongLivedToken` struct actually comes from the bindle client so ... I'm assuming it's supported? This is mostly code that existed prior to this, I just refactored the nested if statements into a match.

Reply:
Yeah, there is a token thing. Right now the authn/z features in bindle are half baked. I have an idea behind making that better but I still need to write it up